core/cas-server-core-services/src/main/java/org/apereo/cas/services/AbstractResourceBasedServiceRegistryDao.java

```diff
@@ -110,7 +114,35 @@ private void initializeRegistry(final Path configDirectory,
 
             LOGGER.info("Watching service registry directory at [{}]", configDirectory);
 
-            this.serviceRegistryConfigWatcher = new ServiceRegistryConfigWatcher(this, eventPublisher);
+            final Consumer<File> onCreate = file -> {
```

Comment:
Possible suggestion for later: might want to move the consumers into a smaller inner/static/private component to make sure this class itself remains as nuclear as possible.